roxie/ccd/ccdkey.cpp

```diff
@@ -639,8 +639,8 @@ class InMemoryDirectReader : public CDirectReaderBase
     {
         if (translator)
         {
-            assertex(_ptr==buf.toByteArray());
-            _ptr = deserializeSource.queryRow();
+            if (_ptr==buf.toByteArray())
```

Comment:
assert fails if the callback is called from within the translator, passing in the source ptr.